server/src/middleware/errorHandler.ts

```diff
@@ -0,0 +1,70 @@
+// SPDX-FileCopyrightText: 2023 The Aalto Grades Developers
+//
+// SPDX-License-Identifier: MIT
+
+import { NextFunction, Request, Response } from 'express';
+import { ValidationError } from 'yup';
+
+export enum HttpCode {
+  BadRequest = 400,
+  Unauthorized = 401,
+  NotFound = 404,
+  InternalServerError = 500,
+  BadGateway = 502
+}
+
+export class CustomError extends Error {
```

Comment:
`CustomError` might be too generic. I suggested `AaltoGradesError` on a whim at one point, though we could probably discuss this and collectively agree on a name.

Reply:
`ApiError` was decided as the name.